src/Microsoft.Diagnostics.Monitoring.WebApi/PrometheusDataModel.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using Microsoft.Diagnostics.Monitoring.EventPipe;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi
+{
+    internal static class PrometheusDataModel
+    {
+        private const char SeperatorChar = '_';
+
+        private static readonly Dictionary<string, string> KnownUnits = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+        {
+            {string.Empty, string.Empty},
+            {"count", string.Empty},
+            {"B", "bytes" },
+            {"MB", "bytes" },
+            {"%", "ratio" },
+        };
+
+        public static string Normalize(string metricProvier, string metric, string unit, double value, out string metricValue)
+        {
+            string baseUnit = null;
+            if ((unit != null) && (!KnownUnits.TryGetValue(unit, out baseUnit)))
+            {
+                baseUnit = unit;
+            }
+            if (string.Equals(unit, "MB", StringComparison.OrdinalIgnoreCase))
+            {
+                value *= 1_000_000; //Note that the metric uses MB not MiB
+            }
+            metricValue = value.ToString(CultureInfo.InvariantCulture);
+
+            StringBuilder builder = new StringBuilder(metricProvier.Length + metric.Length + (baseUnit?.Length ?? 0) + 2);
```

Comment:
I wasn't going to comment, but then I realized if I was confused any maintainers might be in the future: what is the +2? Assuming this has some sort of symbolic meaning, should we replace the magic number with a constant?

Reply:
It's the size for the separator characters. I'll add a comment to reflect that.